catalog/opensearch/catalog-opensearch-source/src/main/java/ddf/catalog/source/opensearch/impl/OpenSearchParserImpl.java

```diff
@@ -338,50 +349,114 @@ private boolean hasParameter(String parameter, List<String> parameters) {
-
-    minY = lat - latDifference;
-    maxY = lat + latDifference;
-
-    if (minY > MIN_LAT && maxY < MAX_LAT) {
-      minX = lon - lonDifference;
-      if (minX < MIN_LON) {
-        minX += MAX_ROTATION;
+  private static Optional<double[]> createBBoxFromPointRadius(
+      final double lonInDegrees, final double latInDegrees, final double searchRadiusInMeters) {
+    final double latDifferenceInDegrees =
+        Math.toDegrees(searchRadiusInMeters / LENGTH_OF_SEMI_MINOR_AXIS_IN_METERS);
+
+    double minLat = latInDegrees - latDifferenceInDegrees; // south
+    double maxLat = latInDegrees + latDifferenceInDegrees; // north
+
+    double minLon; // west
+    double maxLon; // east
+
+    if (minLat > MIN_LAT && maxLat < MAX_LAT) {
+      final double latInRadians = Math.toRadians(latInDegrees);
+
+      final double tanU1 = (1 - FLATTENING) * Math.tan(latInRadians);
+      final double cosU1 = 1 / Math.sqrt((1 + tanU1 * tanU1));
+      final double sigma1 = Math.atan2(tanU1, 0);
+      final double cosSquaredAlpha = 1 - cosU1 * cosU1;
+      final double uSq =
+          cosSquaredAlpha
+              * (LENGTH_OF_SEMI_MAJOR_AXIS_IN_METERS * LENGTH_OF_SEMI_MAJOR_AXIS_IN_METERS
+                  - LENGTH_OF_SEMI_MINOR_AXIS_IN_METERS * LENGTH_OF_SEMI_MINOR_AXIS_IN_METERS)
+              / (LENGTH_OF_SEMI_MINOR_AXIS_IN_METERS * LENGTH_OF_SEMI_MINOR_AXIS_IN_METERS);
+      final double A = 1 + uSq / 16384 * (4096 + uSq * (-768 + uSq * (320 - 175 * uSq)));
+      final double B = uSq / 1024 * (256 + uSq * (-128 + uSq * (74 - 47 * uSq)));
+
+      double cos2sigmaM;
+      double sinSigma;
+      double cosSigma;
+      double deltaSigma;
+
+      double sigma = searchRadiusInMeters / (LENGTH_OF_SEMI_MINOR_AXIS_IN_METERS * A);
+      double sigmaPrime;
+
+      int iterationCount = 0;
+      do {
+        if (iterationCount > MAXIMUM_VICENTYS_FORMULA_ITERATIONS) {
+          LOGGER.debug(
+              "Vincenty's formula failed to converge after {} iterations.",
+              MAXIMUM_VICENTYS_FORMULA_ITERATIONS);
+          return Optional.empty();
+        }
+
+        cos2sigmaM = Math.cos(2 * sigma1 + sigma);
+        sinSigma = Math.sin(sigma);
+        cosSigma = Math.cos(sigma);
+        deltaSigma =
+            B
+                * sinSigma
+                * (cos2sigmaM
+                    + B
+                        / 4
+                        * (cosSigma * (-1 + 2 * cos2sigmaM * cos2sigmaM)
+                            - B
+                                / 6
+                                * cos2sigmaM
+                                * (-3 + 4 * sinSigma * sinSigma)
+                                * (-3 + 4 * cos2sigmaM * cos2sigmaM)));
+        sigmaPrime = sigma;
+        sigma = searchRadiusInMeters / (LENGTH_OF_SEMI_MINOR_AXIS_IN_METERS * A) + deltaSigma;
+
+        iterationCount++;
+      } while (Math.abs(sigma - sigmaPrime) > 1e-12);
```

Comment:
`1e-12` is about 0.006mm. This probably does not need to be that exact, especially because the resulting `double`s are parsed to strings. The maximum `iterationCount` for the situations in the unit tests is only 4.

Reply:
Might want to make this error a little larger to reduce the number of iterations.

Reply:
We could probably calculate what this should be to give us the most accurate decimal in the URL, but I don't think it's that important to spend the time to find the best allowable difference here. This bbox is only calculated once per search, and the iterations are very quick.